frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx

```diff
@@ -367,7 +383,13 @@ export const createClusterKmsResources = (kms: KMSConfig, provider = ProviderNam
       ) && createClusterVaultResources(kms as VaultConfig, clusterKmsResources);
       VaultAuthMethodMapping[vaultConfig.authMethod].supportedEncryptionType.includes(
         KmsEncryptionLevel.STORAGE_CLASS,
-      ) && createCsiVaultResources(kms as VaultConfig, clusterKmsResources, false);
+      ) &&
+        createCsiVaultResources(
+          kms as VaultConfig,
+          clusterKmsResources,
+          false,
+          !encryption.clusterWide,
+        );
```

Comment:
Let's suppose I only checked storage-class checkbox...
Now, when `createClusterKmsResources` is called it will first `createClusterKmsResources` and thus create the advanced resources as well. Then it will call `createCsiVaultResources` with `createAdvancedVaultResource = true` (beacuse !encryption.clusterWide). This will again call the creation of advanced resources right ??